packages/commons/src/logging/index.ts

```diff
@@ -1,2 +1,124 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import winston from "winston";
+import { LoggerConfig } from "../config/commonConfig.js";
+import { bigIntReplacer } from "./utils.js";
+
+export type LoggerMetadata = {
+  serviceName?: string;
+  userId?: string;
+  organizationId?: string;
+  correlationId?: string | null;
+  eventType?: string;
+  eventVersion?: number;
+  streamId?: string;
+};
+
+export const parsedLoggerConfig = LoggerConfig.safeParse(process.env);
+const config: LoggerConfig = parsedLoggerConfig.success
+  ? parsedLoggerConfig.data
+  : {
+      logLevel: "info",
+    };
+
+const logFormat = (
+  msg: string,
+  timestamp: string,
+  level: string,
+  {
+    userId,
+    organizationId,
+    correlationId,
+    eventType,
+    eventVersion,
+    streamId,
+  }: LoggerMetadata,
+  serviceName?: string
+) => {
+  const serviceLogPart = serviceName ? `[${serviceName}]` : undefined;
+  const userLogPart = userId ? `[UID=${userId}]` : undefined;
+  const organizationLogPart = organizationId
+    ? `[OID=${organizationId}]`
+    : undefined;
+  const correlationLogPart = correlationId
+    ? `[CID=${correlationId}]`
+    : undefined;
+  const eventTypePart = eventType ? `[ET=${eventType}]` : undefined;
+  const eventVersionPart = eventVersion ? `[EV=${eventVersion}]` : undefined;
+  const streamIdPart = streamId ? `[SID=${streamId}]` : undefined;
+
+  const firstPart = [timestamp, level.toUpperCase(), serviceLogPart]
+    .filter((e) => e !== undefined)
+    .join(" ");
+
+  const secondPart = [
+    userLogPart,
+    organizationLogPart,
+    correlationLogPart,
+    eventTypePart,
+    eventVersionPart,
+    streamIdPart,
+  ]
+    .filter((e) => e !== undefined)
+    .join(" ");
+
+  return `${firstPart} - ${secondPart} ${msg}`;
+};
+
+export const customFormat = (serviceName?: string) =>
+  winston.format.printf(({ level, message, timestamp, loggerCtx }) => {
+    const clearMessage =
+      typeof message === "object"
+        ? JSON.stringify(message, bigIntReplacer)
+        : message;
+    const lines = clearMessage
+      .toString()
+      .split("\n")
+      .map((line: string) =>
+        logFormat(line, timestamp, level, loggerCtx, serviceName)
+      );
+    return lines.join("\n");
+  });
+
+const getLogger = (serviceName?: string) =>
```

Comment:
If you destructure from logger metadata also `serviceName` (see other comment in the file) you probably don't need it anymore as a param here because `customFormat` would not need it anymore